src/Network/HTTPRequest.php

```diff
@@ -103,120 +105,136 @@ public function get(string $url, array $opts = [], &$redirects = 0)
 		// This seems to confuse curl so that it shows this uncompressed.
 		/// @todo  We could possibly set this value to "gzip" or something similar
-		curl_setopt($ch, CURLOPT_ENCODING, '');
+		$curlOptions[CURLOPT_ENCODING] = '';
 
 		if (!empty($opts['headers'])) {
 			$this->logger->notice('Wrong option \'headers\' used.');
-			@curl_setopt($ch, CURLOPT_HTTPHEADER, $opts['headers']);
+			if (empty($curlOptions[CURLOPT_HTTPHEADER])) {
+				$curlOptions[CURLOPT_HTTPHEADER] = [];
+			}
+			$curlOptions[CURLOPT_HTTPHEADER] = array_merge($opts['headers'], $curlOptions[CURLOPT_HTTPHEADER]);
 		}
 
 		if (!empty($opts['nobody'])) {
-			@curl_setopt($ch, CURLOPT_NOBODY, $opts['nobody']);
+			$curlOptions[CURLOPT_NOBODY] = $opts['nobody'];
 		}
 
-		@curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+		$curlOptions[CURLOPT_CONNECTTIMEOUT] = 10;
 
 		if (!empty($opts['timeout'])) {
-			@curl_setopt($ch, CURLOPT_TIMEOUT, $opts['timeout']);
+			$curlOptions[CURLOPT_TIMEOUT] = $opts['timeout'];
 		} else {
 			$curl_time = $this->config->get('system', 'curl_timeout', 60);
-			@curl_setopt($ch, CURLOPT_TIMEOUT, intval($curl_time));
+			$curlOptions[CURLOPT_TIMEOUT] = intval($curl_time);
 		}
 
 		// by default we will allow self-signed certs
 		// but you can override this
 
 		$check_cert = $this->config->get('system', 'verifyssl');
-		@curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, (($check_cert) ? true : false));
+		$curlOptions[CURLOPT_SSL_VERIFYPEER] = ($check_cert) ? true : false;
 
 		if ($check_cert) {
-			@curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 2);
+			$curlOptions[CURLOPT_SSL_VERIFYHOST] = 2;
 		}
 
 		$proxy = $this->config->get('system', 'proxy');
 
 		if (!empty($proxy)) {
-			@curl_setopt($ch, CURLOPT_HTTPPROXYTUNNEL, 1);
-			@curl_setopt($ch, CURLOPT_PROXY, $proxy);
+			$curlOptions[CURLOPT_HTTPPROXYTUNNEL] = 1;
+			$curlOptions[CURLOPT_PROXY] = $proxy;
 			$proxyuser = $this->config->get('system', 'proxyuser');
 
 			if (!empty($proxyuser)) {
-				@curl_setopt($ch, CURLOPT_PROXYUSERPWD, $proxyuser);
+				$curlOptions[CURLOPT_PROXYUSERPWD] = $proxyuser;
 			}
 		}
 
 		if ($this->config->get('system', 'ipv4_resolve', false)) {
-			curl_setopt($ch, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+			$curlOptions[CURLOPT_IPRESOLVE] = CURL_IPRESOLVE_V4;
 		}
 
-		$s         = @curl_exec($ch);
-		$curl_info = @curl_getinfo($ch);
+		$logger = $this->logger;
 
-		// Special treatment for HTTP Code 416
-		// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/416
-		if (($curl_info['http_code'] == 416) && ($range > 0)) {
-			@curl_setopt($ch, CURLOPT_RANGE, '');
-			$s         = @curl_exec($ch);
-			$curl_info = @curl_getinfo($ch);
-		}
-
-		$curlResponse = new CurlResult($url, $s, $curl_info, curl_errno($ch), curl_error($ch));
+		$onRedirect = function(
+			RequestInterface $request,
+			ResponseInterface $response,
+			UriInterface $uri
+		) use ($logger) {
+			$logger->notice('Curl redirect.', ['url' => $request->getUri(), 'to' => $uri]);
```

Comment:
Are we sure about that? It sounds like it could get out of hand pretty fast in the log.

Reply:
It's the same log-point as before and the parameter `allow_redirect.max` is set to 8 as defined at https://docs.guzzlephp.org/en/stable/request-options.html#allow-redirects